doc/release-notes.md

```diff
@@ -4,36 +4,84 @@ release-notes at release time)
-with this release (built or installed in the same directory as `zcashd`).
-The wallet will not allow the generation of new addresses until this
-confirmation has been performed. It is recommended that after this upgrade,
-that funds tied to preexisting addresses be migrated to newly generated
-addresses so that all wallet funds are recoverable using the emergency
-recovery phrase going forward. If you choose not to migrate funds in this
-fashion, you will continue to need to securely back up the entire `wallet.dat`
-file to ensure that you do not lose access to existing funds; EXISTING FUNDS
-WILL NOT BE RECOVERABLE USING THE EMERGENCY RECOVERY PHRASE UNLESS THEY HAVE
-BEEN MOVED TO A NEWLY GENERATED ADDRESS FOLLOWING THE 4.7.0 UPGRADE.
+The zcashd wallet has been modified to support BIP 39, which describes how to derive the
+wallet's HD seed from a mnemonic phrase.  The mnemonic phrase will be generated on load of
+the wallet, or the first time the wallet is unlocked, and is available via the
+`z_exportwallet` RPC call. All new addresses produced by the wallet are now derived from
+this seed using the HD wallet functionality described in ZIP 32 and ZIP 316. For users
+upgrading an existing Zcashd wallet, it is recommended that the wallet be backed up prior
+to upgrading to the 4.7.0 Zcashd release.
+
+Following the upgrade to 4.7.0, Zcashd will require that the user confirm that they have
+backed up their new emergency recovery phrase, which may be obtained from the output of
+the `z_exportwallet` RPC call. This confirmation can be performed manually using the
+`zcashd-wallet-tool` utility that is supplied with this release (built or installed in the
+same directory as `zcashd`).  The wallet will not allow the generation of new addresses
+until this confirmation has been performed. It is recommended that after this upgrade,
+funds tied to preexisting addresses be migrated to newly generated addresses so that
+all wallet funds are recoverable using the emergency recovery phrase going forward. If you
+choose not to migrate funds in this fashion, you will continue to need to securely back up
+the entire `wallet.dat` file to ensure that you do not lose access to existing funds;
+EXISTING FUNDS WILL NOT BE RECOVERABLE USING THE EMERGENCY RECOVERY PHRASE UNLESS THEY
+HAVE BEEN MOVED TO A NEWLY GENERATED ADDRESS FOLLOWING THE 4.7.0 UPGRADE.
+
+In the case that your wallet previously contained a Sapling HD seed, the mnemonic phrase
+is constructed using the bytes of that seed, such that it is possible to reconstruct 
+keys generated using that legacy seed if you know the mnemonic phrase. HOWEVER, THIS 
+RECONSTRUCTION DOES NOT FOLLOW THE NORMAL PROCESS OF DERIVATION FROM THE MNEMONIC PHRASE.
+Instead, to recover a legacy Sapling key from the mnemonic seed, it is necessary to 
+reconstruct the bytes of the legacy seed by conversion of the mnemonic phrase back to
+its source randomness instead of by hashing as is specified in BIP 39. Only keys and
```

Comment:
```suggestion
Instead, to recover a legacy Sapling key from the emergency recovery phrase, it is necessary
to  reconstruct the bytes of the legacy seed by conversion of the emergency recovery phrase
back to its source randomness instead of by hashing as is specified in BIP 39. Only keys and
```
Unlike the replacements above, the first replacement here is an issue of correctness rather than just tidying up terminology. You *can't* recover the legacy seed from the mnemonic seed, because it's derived using a one-way derivation function. You *can* recover it from the emergency recovery phrase, because (in this case) that's a bijective mapping.